packages/backend/src/ee/models/AiAgentModel.ts

```diff
@@ -299,9 +322,24 @@ export class AiAgentModel {
                             );
                     }
                 }) || [];
-
             const integrations = await Promise.all(integrationPromises);
 
+            let instruction = await this.getAgentLastInstruction({
+                agentUuid: agent.ai_agent_uuid,
+            });
```

Comment:
this should accept a transactions an argument and use it to avoid creating additional db connection,  e.g.: https://github.com/lightdash/lightdash/blob/1dbe0785653000a23511790469f84459d06b38d0/packages/backend/src/models/SpaceModel.ts#L1825